pkgs/development/tools/misc/blisp/default.nix

```
@@ -0,0 +1,51 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  IOKit,
+  testers,
+  blisp,
+}:
+let
+  version = "0.0.3";
+in stdenv.mkDerivation {
+  pname = "blisp";
+  inherit version;
+
+  src = fetchFromGitHub {
+    owner = "pine64";
+    repo = "blisp";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    sha256 = "sha256-m2ETPugSqOVrYNWLYwY5xYuReBImUyfm4jTyAoaqS0k=";
+  };
+
+  nativeBuildInputs = [cmake];
+
+  cmakeFlags = ["-DBLISP_BUILD_CLI=ON"];
+
+  buildInputs = lib.optionals stdenv.isDarwin [IOKit];
+
+  installPhase = ''
+    install -D -t $out/bin tools/blisp/blisp
+  '';
+
+  passthru.tests.version = testers.testVersion {
+    package = blisp;
+    command = "blisp --version";
+    version = "v${version}";
+  };
+
+  meta = with lib; {
+    homepage = "https://github.com/pine64/blisp";
+    description = "In-System Programming (ISP) tool for Bouffalo Labs RISC-V MCUs.";
```

```suggestion
    description = "A In-System Programming (ISP) tool for Bouffalo Labs RISC-V MCUs";
```
In nixpkgs we don't like dots at the end of a package description :shrug: